relay/pkg/metrics/metrics.go

```diff
@@ -0,0 +1,206 @@
+package metrics
+
+import (
+	"context"
+	"time"
+
+	"github.com/ipfs/go-log"
+	"github.com/keep-network/keep-common/pkg/metrics"
+	"github.com/keep-network/tbtc/relay/pkg/btc"
+	"github.com/keep-network/tbtc/relay/pkg/chain"
+	"github.com/keep-network/tbtc/relay/pkg/node"
+)
+
+var logger = log.Logger("relay-metrics")
+
+const (
+	// DefaultChainMetricsTick is the default duration of the
+	// observation tick for chain-related metrics.
+	DefaultChainMetricsTick = 10 * time.Minute
+	// DefaultNodeMetricsTick is the default duration of the
+	// observation tick for node-related metrics.
+	DefaultNodeMetricsTick = 10 * time.Second
+)
+
+// Initialize set up the metrics registry and enables metrics server.
+func Initialize(
+	port int,
+) (*metrics.Registry, bool) {
+	if port == 0 {
+		return nil, false
+	}
+
+	registry := metrics.NewRegistry()
+
+	registry.EnableServer(port)
+
+	return registry, true
+}
+
+// ObserveBtcChainConnectivity triggers an observation process of the
+// btc_chain_connectivity metric.
+func ObserveBtcChainConnectivity(
+	ctx context.Context,
+	registry *metrics.Registry,
+	btcHandle btc.Handle,
+	tick time.Duration,
+) {
+	input := func() float64 {
+		_, err := btcHandle.GetBlockCount()
+
+		if err != nil {
+			return 0
+		}
+
+		return 1
+	}
+
+	observe(
+		ctx,
+		"btc_chain_connectivity",
+		input,
+		registry,
+		validateTick(tick, DefaultChainMetricsTick),
+	)
+}
+
+// ObserveHostChainConnectivity triggers an observation process of the
+// host_chain_connectivity metric.
+func ObserveHostChainConnectivity(
+	ctx context.Context,
+	registry *metrics.Registry,
+	hostChain chain.Handle,
+	tick time.Duration,
+) {
+	input := func() float64 {
+		_, err := hostChain.GetBestKnownDigest()
+
+		if err != nil {
+			return 0
+		}
+
+		return 1
+	}
+
+	observe(
+		ctx,
+		"host_chain_connectivity",
+		input,
+		registry,
+		validateTick(tick, DefaultChainMetricsTick),
+	)
+}
+
+// ObserveBlockForwarding triggers an observation process of the
+// block_forwarding metric.
+func ObserveBlockForwarding(
+	ctx context.Context,
+	registry *metrics.Registry,
+	nodeStats *node.Stats,
+	tick time.Duration,
+) {
+	input := func() float64 {
+		isEnabled := nodeStats.BlockForwardingEnabled()
+
+		if !isEnabled {
+			return 0
+		}
+
+		return 1
+	}
+
+	observe(
+		ctx,
+		"block_forwarding",
```

Comment:
Thoughts on `block_forwarding_status`/`block_forwarding_state`? Just `block_forwarding` is a little unclear to me if it's just 0/1 or maybe some counter of blocks processed.

Reply:
Oh, maybe `_active`? I think I've just realized that this metric will be updated based on the relay auction result.

Reply:
Yep, good catch. See https://github.com/keep-network/tbtc/commit/4b0a178cb5e6ce2fc02e92a54717ab42c3d6a8ab